.github/workflows/build.yml

```diff
@@ -1,18 +1,29 @@
-name: Build Podiscover on Docker
+name: Build and Publish Docker Image
 
 on:
-  pull_request:
-    branches: [main]
   push:
     branches: [main]
 
 jobs:
   build:
-    name: Build Docker Image for Podiscover
     runs-on: ubuntu-latest
+
     steps:
-      - name: Checkout
+      - name: Checkout repository
         uses: actions/checkout@v2
 
-      - name: Build Docker Image
-        run: docker build -t podiscover:${{github.sha}} .
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```

Comment:
do we need to add them to env or where something similar?

Reply:
It should be added to Secrets for Actions, here how I did: 

<img width="1275" alt="Screenshot 2023-04-20 at 09 06 41" src="https://user-images.githubusercontent.com/3473393/233272850-77b6b1db-7b18-464e-b1db-32108b61c036.png">